tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -0,0 +1,293 @@
+/// \file ROOT/RNTupleProcessor.hxx
+/// \ingroup NTuple ROOT7
+/// \author Florine de Geus <florine.de.geus@cern.ch>
+/// \date 2024-03-26
+/// \warning This is part of the ROOT 7 prototype! It will change without notice. It might trigger earthquakes. Feedback
+/// is welcome!
+
+/*************************************************************************
+ * Copyright (C) 1995-2024, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT7_RNTupleProcessor
+#define ROOT7_RNTupleProcessor
+
+#include <ROOT/REntry.hxx>
+#include <ROOT/RError.hxx>
+#include <ROOT/RNTupleDescriptor.hxx>
+#include <ROOT/RNTupleModel.hxx>
+#include <ROOT/RNTupleUtil.hxx>
+#include <ROOT/RPageStorage.hxx>
+#include <ROOT/RSpan.hxx>
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace ROOT {
+namespace Experimental {
+
+/// Helper type representing the name and storage location of an RNTuple.
+struct RNTupleSourceSpec {
+   std::string fName;
+   std::string fStorage;
+
+   RNTupleSourceSpec() = default;
```

Comment:
I adopted this from what's used for `RNTupleReader::OpenFriends`:

https://github.com/root-project/root/blob/f53aaf03beb72a0d0548e9d2b23a317600b76184/tree/ntuple/v7/inc/ROOT/RNTupleReader.hxx#L137-L145

It would make sense to make this a common struct if this ends up in the complete version of the processor.